tests/integration/aof-multi-part.tcl

```diff
@@ -0,0 +1,1065 @@
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.8.base.rdb seq 8 type b} 
+                {file appendonly.aof.4.incr.aof seq 4 type i}
+            }
+
+            r set k2 v2
+            r debug loadaof
+
+            # Both k1 and k2 lost
+            assert_equal 0 [r exists k1]
+            assert_equal 0 [r exists k2]
+
+            r config set rdb-key-save-delay 0
+            waitForBgsave r
+
+            after 2000 ; # Make sure AOFRW was scheduled
+
+            waitForBgrewriteaof r
+
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.9.base.rdb seq 9 type b} 
+                {file appendonly.aof.5.incr.aof seq 5 type i}
+            }
+
+            r set k3 v3
+            r debug loadaof
+            assert_equal v3 [r get k3]
+        }
+
+        test "AOF will trigger limit when AOFRW fails many times" {
+            r config set rdb-key-save-delay 10000000
+            # Let us trigger AOFRW easily
+            r config set auto-aof-rewrite-percentage 1
+            r config set auto-aof-rewrite-min-size 1mb
+
+            # Let AOFRW fail two times, this will trigger AOFRW limit
+            r bgrewriteaof
+            set fork_child_pid [get_child_pid 0]
+            exec kill -9 $fork_child_pid
+            waitForBgrewriteaof r
+
+            r bgrewriteaof
+            set fork_child_pid [get_child_pid 0]
+            exec kill -9 $fork_child_pid
+            waitForBgrewriteaof r
+
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.9.base.rdb seq 9 type b} 
+                {file appendonly.aof.5.incr.aof seq 5 type i}
+                {file appendonly.aof.6.incr.aof seq 6 type i}
+                {file appendonly.aof.7.incr.aof seq 7 type i}
+            }
+
+            set orig_size [r dbsize]
+            set load_handle0 [start_write_load $master_host $master_port 10]
+
+            wait_for_condition 50 100 {
+                [r dbsize] > $orig_size
+            } else {
+                fail "No write load detected."
+            }
+
+            # Make sure we have limit log
+            wait_for_condition 1000 500 {
+                [count_log_message 0 "triggered the limit"] == 1
+            } else {
+                fail "aof rewrite did trigger limit"
+            }
+
+            # Wait 1 sec
+            after 1000
+
+            # No new INCR AOF be created
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.9.base.rdb seq 9 type b} 
+                {file appendonly.aof.5.incr.aof seq 5 type i}
+                {file appendonly.aof.6.incr.aof seq 6 type i}
+                {file appendonly.aof.7.incr.aof seq 7 type i}
+            }
+
+            # Turn off auto rewrite
+            r config set auto-aof-rewrite-percentage 0
+            r config set rdb-key-save-delay 0
```

Comment:
maybe assert here that aof_rewrite_in_progress is 0 too?
i.e. just to make it clear that the next bgrewriteaof is gonna be the one that succeeds.

Reply:
OK